### PR TITLE
VMware: Unit test for vmware_guest

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
test/units/modules/cloud/vmware/__init__.py
test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json
test/units/modules/cloud/vmware/test_vmware_guest.py

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte @jeking3 Could you please review this ?

**Review by pdellaert — APPROVED** [review]

**Comment by jeking3** [comment]
I'd have preferred mock (MagicMock), it seems to be used pretty widely.  But it looks like the same basic thing can be done with pytest.  I'd leave you with one question and one answer... how do you know when you are done testing?  Without branch coverage numbers, you can't know.  Is the ansible unit test framework capturing branch coverage of unit tests?

File: test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        {
+            "name": "sample_vm_001",
+            "hostname": "esxi01.example.com",
+            "username": "administrator@vsphere.local",
+            "password": "Secret@123$%",
+            "validate_certs": "True"
+        },
+        {
+            "failed": "True",
+            "msg": "Unknown error while connecting to vCenter or ESXi API"
```
Comment (jeking3) [inline test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json:12]: My definition of unit testing is a test that isolates the module being tested.  This means attempting to isolate it as much as possible.  I'm not seeing how the interaction with pyVmomi is being mocked or patched.  Without patching pyVmomi responses, it won't be possible to unit test failure paths.

**Comment by Akasurde** [comment]
> I'd have preferred mock (MagicMock), it seems to be used pretty widely. But it looks like the same 

Yes, we can use MagicMock as well.

> basic thing can be done with pytest. I'd leave you with one question and one answer... how do you know when you are done testing? 

This is just starting of this effort, I am planning to add more test cases to cover all branches. This PR will lay ground for future work.

> Without branch coverage numbers, you can't know. Is the ansible unit test framework capturing branch coverage of unit tests?

We don't have number as such to know how much we are covering but we can definitely start working towards it. Also, Ansible unit test framework does not give you branch coverage details :(

File: test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json (hunk 1 above)
Comment (Akasurde) [reply test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json:12]: Like I said this is just starting efforts, and will continue to add more testcases based on pyVmomi mocking.

**Review by jeking3 — APPROVED** [review]
I'd prefer to see magicmock instead, but we're moving in the right direction here.  Let's keep moving forward!

**Pull request merged**